### PR TITLE
Add timing tests for deserialization + move validation from deserialization to separate function #177

### PR DESCRIPTION
**Pull request by gijsvl**

### CODE REVIEW

File: zkabacus-crypto/src/proofs.rs [hunk 1]
```diff
@@ -810,4 +812,73 @@ mod tests {
...
+            .unblind(old_pt_bf);
+
+        // Save a copy of the nonce...
+        let nonce = *old_state.nonce();
+
+        let context = Context::new(b"pay proof verify test");
+
+        // Form proof.
+        let param_ser = bincode::serialize(&params).unwrap();
+        let start = Instant::now();
```
Comment (marsella) [inline zkabacus-crypto/src/proofs.rs:853]: Nonblocking: You might consider giving more meaningful names to all the `start` timers. Maybe `config_deserialize_timer` here, `proof_creation_timer` on L855, etc.

File: zkchannels-crypto/src/proofs/range.rs [hunk 2]
```diff
@@ -102,6 +69,16 @@ impl RangeConstraintParameters {
     pub fn public_key(&self) -> &PublicKey<1> {
         &self.public_key
     }
+
+    /// Verify the signatures inside the RangeConstraintParameters are valid signatures on the appropriate scalars
+    pub fn validate(&self) -> Result<(), String> {
+        for (i, sig) in self.digit_signatures.iter().enumerate() {
+            if !sig.verify(&self.public_key, &Scalar::from(i as u64).into()) {
+                return Err("The signatures in the RangeConstraintParameters must be valid signatures on the appropriate scalars".to_string());
```
Comment (marsella) [inline zkchannels-crypto/src/proofs/range.rs:77]: Thought: It looks like we use string errors throughout the library, so I would not recommend changing it, but at some point we may find it useful to make named error types. We don't match on the errors anywhere that we use this library, but it seems like a professional, Rusty thing to do.

**Review by marsella — APPROVED** [review]
This looks good! One comment, to be changed at your discretion, and one thought.

File: zkchannels-crypto/src/proofs/range.rs (hunk 2 above)
Comment (indomitableSwan) [reply zkchannels-crypto/src/proofs/range.rs:77]: Although on that note, I did, on Mukund's input, create named errors in revlock.rs and found it useful for testing.

**Pull request merged**